packages/parser/src/parser.ts

```diff
@@ -76,14 +76,14 @@ function getLib(compilerOptions: CompilerOptions): Lib[] {
 }
 
 function parse(
-  code: string,
+  code: string | ts.SourceFile,
   options?: ParserOptions,
 ): ParseForESLintResult['ast'] {
   return parseForESLint(code, options).ast;
 }
 
 function parseForESLint(
-  code: string,
+  code: string | ts.SourceFile,
```

Comment:
FYI @Quramy - is there anything else you'd need/want done to help with the use case described in #774?

cc @mhartington - you were showing me using Quramy's language service plugin. I'm betting this will speed it up ever so slightly (though I'd be surprised if file parsing was a major bottleneck). ✨